### PR TITLE
Feature/router

### PR DESCRIPTION
**Pull request by aokumasan**
## Description

- Add nifcloud_router resource.
    - Implemented #9

## How Has This Been Tested?

- [x]  Buld with `make install` command, and Apply example.

```:console
make install
cd examples/router
terraform init -get-plugins=false
terraform plan
terraform apply
```

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

### CODE REVIEW

File: nifcloud/resources/router/flattener.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+			} else {
+				ni["network_id"] = nifcloud.StringValue(n.NetworkId)
+				ni["network_name"] = nifcloud.StringValue(n.NetworkName)
+				ni["ip_address"] = nifcloud.StringValue(n.IpAddress)
+				ni["dhcp"] = nifcloud.BoolValue(n.Dhcp)
+				ni["dhcp_options_id"] = nifcloud.StringValue(n.DhcpOptionsId)
+				ni["dhcp_config_id"] = nifcloud.StringValue(n.DhcpConfigId)
+			}
+
+			log.Println("ni", ni)
```
Comment (fuku2014) [inline nifcloud/resources/router/flattener.go:131]: Is this log for debugging?

File: nifcloud/resources/router/schema.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+			Computed:    true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(1, 15),
+				validation.StringMatch(regexp.MustCompile(`^[0-9a-zA-Z]+$`), ""),
+			),
+		},
+		"nat_table_association_id": {
+			Type:        schema.TypeString,
+			Description: "The ID of the NAT table association.",
+			Optional:    true,
```
Comment (fuku2014) [inline nifcloud/resources/router/schema.go:72]: this does not need to be defined.

File: nifcloud/resources/router/flattener.go (hunk 1 above)
Comment (aokumasan) [reply nifcloud/resources/router/flattener.go]: sorry... fixed!
9a50eb2711f5573c151e2a39100915dfa8784b06

File: nifcloud/resources/router/schema.go (hunk 2 above)
Comment (aokumasan) [reply nifcloud/resources/router/schema.go]: fixed! f6003e8bc24d4d703f8704d1f50e12ad4e31abfd

File: nifcloud/acc/testdata/router_update.tf [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  region = "jp-east-2"
+}
+
+resource "nifcloud_router" "basic" {
+  name              = "%supd"
+  description       = "memo-upd"
+  availability_zone = "east-21"
+  accounting_type   = "2"
+  type              = "small"
+  security_group    = nifcloud_security_group.basic.group_name
```
[the remark this answers is not in the document]
Comment (aokumasan) [reply nifcloud/acc/testdata/router_update.tf:11]: Fixed to update the more fields.
ad076433c814df5880037726e9875f4cd56d3966

**Comment by fuku2014** [comment]
LGTM😻

**Pull request merged**